### ISSUE TITLE
Add lr scheduler for optimizer

### ISSUE THREAD

**Issue by wanchaol** [pytorch/torchtitan#4]
see https://github.com/pytorch-labs/torchtrain/blob/main/train.py#L83
